### PR TITLE
DIV-6145-PFE-updated-to-display-general-orders

### PR DESCRIPTION
**Pull request by madjava**
# Description

https://tools.hmcts.net/jira/browse/DIV-6145

Updated PFE to display General Orders as part of document listed if available.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/core/utils/viewHelper.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * @param session Session data containing d8 property, an array of documents
+ * @returns {*} List of generated documents or an empty if none
+ */
+const getDownloadableFiles = session => {
+  return createUris(session.d8, docConfig);
+};
+
+module.exports = {
+  getDownloadableFiles
+};
```
Comment (madjava) [inline app/core/utils/viewHelper.js:26]: Created a new file to add reusable functionality for getting downloadable documents

File: app/core/utils/viewHelper.test.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      const generalOrderDocuments = getOnlyFileType(fileTypes, 'GeneralOrder');
+
+      expect(fileTypes).to.have.lengthOf(expectedDocumentsSize);
+      expect(fileTypes).to.include('dpetition');
+      expect(fileTypes).to.include('DeemedServiceRefused');
+      expect(fileTypes).to.include('GeneralOrder');
+      expect(generalOrderDocuments).to.have.lengthOf(expectedGeneralOrderDocumentsSize);
+    });
+  });
+});
```
Comment (madjava) [inline app/core/utils/viewHelper.test.js:114]: Test suite with mock objects to verify documents exist and expected behaviour when there arent any documents in list

File: app/steps/amendment-explanatory-page/content.json [hunk 3]
```diff
@@ -19,7 +19,7 @@
           "amendedApplicationInfoPara1": "You recently got in touch with the divorce centre asking to change parts of your application. If you did not make this request please contact the courts immediately.",
           "amendedApplicationInfoPara2": "You’ll only be able to amend and resubmit your case as long as your husband of wife has not yet responded (acknowledgement of service).",
           "amendedApplicationInfoPara3": "The fee for an amended application is £95.",
-          "amendedApplicationInfoPara4": "Once the amended application has been submitted, it will be checked by the courts before being issued to the petitioner’s husband/wife so they can respond.",
+          "amendedApplicationInfoPara4": "Once the amended application has been submitted, it will be checked by the courts before being issued to the petitioner’s {{ divorceWho }} so they can respond.",
```
Comment (madjava) [inline app/steps/amendment-explanatory-page/content.json:22]: Updated this to use already existing `divorceWho` variable in UI which read better, its done every where on PFE

File: app/steps/amendment-explanatory-page/content.json [hunk 4]
```diff
@@ -42,7 +42,8 @@
             "deemedAsServedGranted": "Deemed service order",
             "dispenseWithServiceGranted": "Order to dispense",
             "DeemedServiceRefused": "Deemed service refusal",
-            "DispenseWithServiceRefused": "Dispensed service refusal"
+            "DispenseWithServiceRefused": "Dispensed service refusal",
+            "GeneralOrder": "General Order"
```
Comment (madjava) [inline app/steps/amendment-explanatory-page/content.json:46]: Added `General Order` to list of displayable documents

File: app/steps/amendment-explanatory-page/content.json [hunk 5]
```diff
@@ -89,7 +90,8 @@
             "deemedAsServedGranted": "[CY] Deemed service order",
             "dispenseWithServiceGranted": "[CY] Order to dispense",
             "DeemedServiceRefused": "[CY] Deemed service refusal",
-            "DispenseWithServiceRefused": "[CY] Dispensed service refusal"
+            "DispenseWithServiceRefused": "[CY] Dispensed service refusal",
+            "GeneralOrder": "[CY] General Order"
```
Comment (madjava) [inline app/steps/amendment-explanatory-page/content.json:94]: Same for CY

File: app/steps/amendment-explanatory-page/index.js [hunk 6]
```diff
@@ -40,12 +39,7 @@ module.exports = class AwaitingAmend extends Step {
   }
 
   getDownloadableFiles(session) {
-    const docConfig = {
-      documentNamePath: config.document.documentNamePath,
-      documentWhiteList: config.document.filesWhiteList
-    };
-
-    return createUris(session.d8, docConfig);
```
Comment (madjava) [inline app/steps/amendment-explanatory-page/index.js:48]: Move functionality to common location as its now used in more than one place

File: app/steps/amendment-explanatory-page/index.test.js [hunk 7]
```diff
@@ -19,8 +20,14 @@ let underTest = {};
...
 const BASE_PATH = '/';
 let session = {};
+let serviceCentre = {};
 let postBody = {};
 
+const buildAwaitingAmendSession = () => {
+  const oneSecond = 1000;
+  return Object.assign({}, mockAwaitingAmendSession, { expires: Date.now() + oneSecond });
+};
+
```
Comment (madjava) [inline app/steps/amendment-explanatory-page/index.test.js:30]: Co-locating common functionality for creating mock session in test suite

File: app/steps/amendment-explanatory-page/index.test.js [hunk 8]
```diff
@@ -63,63 +67,81 @@ describe(modulePath, () => {
...
+    it('should have one \'dpetition\' label in template view', done => {
+      const dpetitionFileLabel = getTemplateFileLabel(content, 'dpetition');
+      testExistence(done, agent, underTest, dpetitionFileLabel);
+    });
+
+    it('should have one \'General Order\' label in template view', done => {
+      const generalOrderFileLabel = getTemplateFileLabel(content, 'GeneralOrder');
+      testExistence(done, agent, underTest, generalOrderFileLabel);
+    });
+  });
```
Comment (madjava) [inline app/steps/amendment-explanatory-page/index.test.js:87]: Testing the General Order is also available

File: config/default.yaml [hunk 9]
```diff
@@ -205,3 +205,4 @@ document:
     - 'd8petition'
     - 'DispenseWithServiceRefused'
     - 'DeemedServiceRefused'
+    - 'GeneralOrder'
```
Comment (madjava) [inline config/default.yaml:208]: Adding GeneralOrder to configuration white list

File: test/end-to-end/paths/serviceApplicationNotApproved.js [hunk 10]
```diff
@@ -14,6 +14,6 @@ Scenario('Service application not approved screen with expected information', as
   I.see(content.whichSituation);
 
   I.see('Deemed service refusal (PDF)', '.govuk-link');
+  I.see('General Order (PDF)', '.govuk-link');
   I.dontSee('Dispense with service refusal (PDF)', '.govuk-link');
-
```
Comment (madjava) [inline test/end-to-end/paths/serviceApplicationNotApproved.js:18]: Updating end-to-end test to validate General order documents is visible

File: test/util/helpers.js [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+const getOnlyFileType = (fileTypes, typeName) => {
+  return filter(fileTypes, fileType => {
+    return fileType === typeName;
+  });
+};
+
+module.exports = {
+  getTemplateFileLabel,
+  getOnlyFileType
+};
```
Comment (madjava) [inline test/util/helpers.js:18]: Co-locating common testing functionality

File: app/steps/service-application-not-approved/index.test.js [hunk 12]
```diff
@@ -62,11 +73,24 @@ describe(modulePath, () => {
...
+      it('should have one \'DeemedServiceRefused\' label in template view', done => {
+        const deemedServiceRefusedFileLabel = getTemplateFileLabel(content, 'DeemedServiceRefused');
+        testExistence(done, agent, underTest, deemedServiceRefusedFileLabel);
+      });
+
+      it('should have two \'GeneralOrder\' labels in template view', done => {
+        // eslint-disable-next-line max-nested-callbacks
+        testCustom(done, agent, underTest, [], response => {
+          const numberOfItems = 2;
+          expect(getGeneralOrderOccurrencesInPage(response)).to.equal(numberOfItems);
```
Comment (Dizrupto) [inline app/steps/service-application-not-approved/index.test.js:86]: Nice one :)

**Review by Dizrupto — DISMISSED** [review]

File: app/steps/amendment-explanatory-page/content.json (hunk 5 above)
Comment (yarpo) [reply app/steps/amendment-explanatory-page/content.json:94]: Is your ticket labeled with `NeedsWelsh`?

Comment (madjava) [reply app/steps/amendment-explanatory-page/content.json:94]: @yarpo No its not, would ask Jeremy to add label

Comment (madjava) [reply app/steps/amendment-explanatory-page/content.json:94]: @yarpo Pointed it out to Jeremy, he's added appropriate label for Welsh

File: app/steps/amendment-explanatory-page/content.json [hunk 13]
```diff
@@ -41,8 +41,9 @@
             "rejectionDnRefusalOrder": "Refusal Order",
             "deemedAsServedGranted": "Deemed service order",
             "dispenseWithServiceGranted": "Order to dispense",
-            "DeemedServiceRefused": "Deemed service refusal",
-            "DispenseWithServiceRefused": "Dispensed service refusal"
+            "deemedServiceRefused": "Deemed service refusal",
+            "dispenseWithServiceRefused": "Dispensed service refusal",
+            "generalOrder": "General Order"
```
Comment (madjava) [inline app/steps/amendment-explanatory-page/content.json:46]: Updated to use camel case naming convention

File: package.json [hunk 14]
```diff
@@ -181,7 +181,9 @@
     "minimist": "^1.2.5",
     "serialize-javascript": ">=2.1.1",
     "tunnel-agent": "^0.6.0",
-    "yargs-parser": "15.0.1"
+    "yargs-parser": "15.0.1",
+    "bl": ">=3.0.1",
+    "dot-prop": ">=5.1.1"
```
Comment (madjava) [inline package.json:186]: Fixing vulnerability issues to allow build go green

**Review by yarpo — DISMISSED** [review]

File: app/steps/amendment-explanatory-page/template.html [hunk 15]
```diff
@@ -9,7 +9,10 @@
 
 {% block breadcrumbs %}
 {% set html %}
-{{ common.feedback | safe }}
+  {{ common.feedback | safe }}
+   {% if fields.featureToggles.value.ft_welsh == "true" %}
+    {{ common.languageToggle | safe }}
+   {% endif %}
```
Comment (madjava) [inline app/steps/amendment-explanatory-page/template.html:15]: Finally figured out how to add the CY toggle on the page layout. Updating the newly created screens with the same

File: app/steps/service-application-not-approved/template.html [hunk 16]
```diff
@@ -8,7 +8,10 @@
 {% block breadcrumbs %}
 
 {% set html %}
-{{ common.feedback | safe }}
+  {{ common.feedback | safe }}
+   {% if fields.featureToggles.value.ft_welsh == "true" %}
+    {{ common.languageToggle | safe }}
+  {% endif %}
```
Comment (madjava) [inline app/steps/service-application-not-approved/template.html:14]: Ditto

File: app/views/includes/sideMenu.html [hunk 17]
```diff
@@ -30,11 +30,11 @@ <h4 class="govuk-heading-xs govuk-!-margin-0">{{ session.court.serviceCentre.ser
   </ul>
 
   {% if (session.downloadableFiles.length > 0) %}
-  <h3 class="govuk-heading-m">{{ content.downloadDocumentsHeading }}</h3>
+  <h3 class="govuk-heading-s">{{ content.downloadDocumentsHeading }}</h3>
   {{ documentList(session.downloadableFiles, content.files) }}
   {% endif %}
 
-  <h3 class="govuk-heading-m" id="subsection-title">{{ content.guidanceHeading }}</h3>
+  <h3 class="govuk-heading-s" id="subsection-title">{{ content.guidanceHeading }}</h3>
```
Comment (madjava) [inline app/views/includes/sideMenu.html:37]: Noticed the font sizes here where a bigger that the usual, so updated them.

**Review by yarpo — APPROVED** [review]

**Review by Dizrupto — APPROVED** [review]

**Pull request merged**